metadata-ingestion/src/datahub/ingestion/transformer/add_dataset_dataproduct.py

```diff
@@ -0,0 +1,141 @@
+import logging
+from typing import Callable, Dict, List, Optional, Union
+
+import pydantic
+
+from datahub.configuration.common import (
+    ConfigurationError,
+    KeyValuePattern,
+    TransformerSemanticsConfigModel,
+)
+from datahub.configuration.import_resolver import pydantic_resolve_key
+from datahub.emitter.mce_builder import Aspect
+from datahub.emitter.mcp import MetadataChangeProposalWrapper
+from datahub.ingestion.api.common import PipelineContext
+from datahub.ingestion.transformer.dataset_transformer import (
+    DatasetDataproductTransformer,
+)
+from datahub.metadata.schema_classes import MetadataChangeProposalClass
+from datahub.specific.dataproduct import DataProductPatchBuilder
+
+logger = logging.getLogger(__name__)
+
+
+class AddDatasetDataProductConfig(TransformerSemanticsConfigModel):
```

Comment:
why does this inherit from `TransformerSemanticsConfigModel` - I don't think it should